### ISSUE TITLE
QRCode Label Font Size

### ISSUE THREAD

**Issue by tcohen6** [endroid/qr-code#379]
Endroid QRCode Library - I am unable to change the font size of my label in the most recent version. I saw "setLabelFontSize" in the documentation for past versions but no equivalent in the current documentation. How can I make the label font size larger? Code is below. Thanks

use Endroid\QrCode\QrCode;
use Endroid\QrCode\Writer\PngWriter;
use Endroid\QrCode\ErrorCorrectionLevel\ErrorCorrectionLevelHigh;
use Endroid\QrCode\Color\Color;
use Endroid\QrCode\Logo\Logo;
use Endroid\QrCode\Label\Label;
use Endroid\QrCode\Label\Label\Font;
use Endroid\QrCode\Label\Font\NotoSans;
$qr = QrCode::create("my qr code text")
  ->setErrorCorrectionLevel(new ErrorCorrectionLevelHigh())
  ->setSize(400)
  ->setMargin(20)
  ->setForegroundColor(new Color(0, 0, 0))
  ->setBackgroundColor(new Color(255, 255, 255));
  $label = Label::create("my webiste")
    ->setTextColor(new Color(0, 0, 0));
  $writer = new PngWriter();
$result = $writer->write($qr, $logo, $label);

**Comment by endroid** [comment]
Hi @tcohen6 the font can be passed as the second parameter of the Label constructor or via the setFont method.

**Comment by tcohen6** [comment]
Thx

-----------------------------------

**Comment by adeelahmed233** [comment]
I am not able to change font size of label. I am creating label without bundle. I did not find it in README. Here is the code I am using

use Endroid\QrCode\Color\Color;
use Endroid\QrCode\Encoding\Encoding;
use Endroid\QrCode\ErrorCorrectionLevel\ErrorCorrectionLevelLow;
use Endroid\QrCode\QrCode;
use Endroid\QrCode\Label\Label;
use Endroid\QrCode\Logo\Logo;
use Endroid\QrCode\RoundBlockSizeMode\RoundBlockSizeModeMargin;
use Endroid\QrCode\Writer\PngWriter;

$writer = new PngWriter();

// Create QR code
$qrCode = QrCode::create(''.$data["realqrcode"].'')
    ->setEncoding(new Encoding('UTF-8'))
    ->setErrorCorrectionLevel(new ErrorCorrectionLevelLow())
    ->setSize($qr_size_doc)
    ->setMargin($qrmargin)
    ->setRoundBlockSizeMode(new RoundBlockSizeModeMargin())
    ->setForegroundColor(new Color($fr, $fg, $fb))
    ->setBackgroundColor(new Color($br, $bg, $bb));

// Create generic logo
$logo = Logo::create(__DIR__.'/logo.png')
    ->setResizeToWidth($qrlogosize);

// Create generic label
$label = Label::create(''.$qr_label_doc.'')
    ->setTextColor(new Color($lr, $lg, $lb))
    **->setFont(new NotoSans(20));**

I am new to object oriented programming and did not understand words _Label constructor_ or _setFont method_. I would appreciate if you could provide sample code for the bold line of code above.

You have made the great library. Working great so far. Thank you very much!

**Comment by tcohen6** [comment]
Solved thank you

-----------------------------------

**Comment by adeelahmed233** [comment]
Can someone please provide sample code to change font size without builder?

**Comment by endroid** [comment]
@adeelahmed233 I think the code you posted should be correct. Can you use the DebugWriter and share the output? That should show the effectuated font size.

**Comment by adeelahmed233** [comment]
Sorry I dont know about DebugWriter. Will look into it. Here is the error I get in the browser while using the code

Fatal error: Uncaught Error: Class "NotoSans" not found in C:\xampp\htdocs\verify.sealqr.com\mycode\generate_qr.php:50

here are lines 48,49 and 50 (bold) of generate_qr.php file

$label = Label::create(''.$qr_label_doc.'')
    ->setTextColor(new Color($lr, $lg, $lb))
    **->setFont(new NotoSans(2));**

If I remove line 50, the code works fine.

**Comment by endroid** [comment]
That looks like you didn't include a use statement for the class.

**Comment by adeelahmed233** [comment]
You are right. I added _use Endroid\QrCode\Label\Font\NotoSans;_ and the issue has been solved. Thanks a lot. the issue can be closed.
